### PR TITLE
Add special note for EDD and EF

### PR DESCRIPTION
**Pull request by rkac-bw**
## 📔 Objective

Add specific info in regards to implementing EDD process for Entity framework Only Databases with Code Migrations (Self host for example)

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation
  team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Comment by Hinton** [comment]
Is this needed? To my knowledge we don't use EF migrations for cloud and unified has no requirement for EDD.

File: docs/contributing/database-migrations/edd.mdx [hunk 1]
```diff
@@ -308,6 +308,26 @@ Prior to merging a PR please ensure that the database changes run well on the cu
 version. We currently do not have an automated test suite for this and it’s up to the developers to
 ensure their database changes run correctly against the currently released version.
 
+## EF-Only Databases with Code Migrations
```
Comment (withinfocus) [inline docs/contributing/database-migrations/edd.mdx:311]:
```suggestion
## EF-only databases with code migrations
```

**Comment by withinfocus** [comment]
> Is this needed? To my knowledge we don't use EF migrations for cloud and unified has no requirement for EDD.

EF migrations are used for SM in the cloud right? I'd rather be consistent in our approach nonetheless though. It would be ideal to see a PR that includes EDD-prescribed changes both in a traditional DbUp migration script as well as an EF one.

**Review by withinfocus — CHANGES_REQUESTED** [review]
Your lint is failing -- make sure that's running locally for you per [the setup guide](https://github.com/bitwarden/contributing-docs?tab=readme-ov-file#install).

**Comment by Hinton** [comment]
No, SM uses regular migrations since using the EF framework for half our migration and hand written for the rest would be very difficult to understand.

For the delete scenario we should be fine having EF migrations that immediately delete the value. We will lose the ability to roll back though but for a delete that should be fine as we never rollback self-hosted instances.

There are more complex scenarios like column rename that requires some juggling with EF. And requires 3 deployments + 1 data migration.

1. Add column `new`, update code to write both `old` and `new`. <- DB & code deploy.
2. Run DB migration copying all `old` to `new`. <- DB
3. Removing code references to `old`. <- Code
4. Remove `old` from database. <- DB

File: docs/contributing/database-migrations/edd.mdx [hunk 2]
```diff
@@ -308,6 +308,26 @@ Prior to merging a PR please ensure that the database changes run well on the cu
...
+
+For EF only databases that use code as migrations (self-host), migrations will run before new code is deployed and therefore before the model that EF works off of has changed, since that is in the code.
+
+Columns that are referenced in queries in the database have to be deleted over 2 releases. Here's the process:
+
+1. Before the first release, delete the property in the C# class.
+
+2. Keep that column in as a shadow property. Shadow properties are properties that are defined only in the EF model but are not defined in the C# classes. This can be particularly useful when you're transitioning columns in the database schema without immediately reflecting those changes in the code.
+
```
Comment (eliykat) [inline docs/contributing/database-migrations/edd.mdx:320]: How do I do this? I had a look at https://learn.microsoft.com/en-us/ef/core/modeling/shadow-properties but it wasn't obvious how that maps to our implementation.

**Comment by eliykat** [comment]
Is there a consensus here? I need to drop some columns currently - which approach should I be using?

Regarding the PR itself - it would be useful to have a timeline showing both database frameworks side-by-side. e.g.

---

Example: dropping a column

| C# Entity | MSSQL | EF |
| -- |-- | -- |
| No change | Give the parameter a default value in all stored procedures | No change |
| Release |
| Delete the property (migration) | No change | Configure the column as a shadow property (no migration) |
| Release |
| No change | Drop the column | Delete the shadow property (migration) |

---

Maybe there is a better way to represent this.

**Review by eliykat — CHANGES_REQUESTED** [review]

**Comment by withinfocus** [comment]
As discussed we're got gonna EDD EF at this time.

**Pull request closed without merging**